frontend/js/src/explore/fresh-releases/FreshReleases.tsx

```diff
@@ -217,29 +231,66 @@ export default function FreshReleases() {
         const releaseTags = Array.from(uniqueReleaseTagsSet);
         releaseTags.sort();
 
-        setReleases(cleanReleases);
-        setFilteredList(cleanReleases);
-        setAllFilters({
-          releaseTypes,
-          releaseTags,
-        });
-        setIsLoading(false);
+        return {
+          data: {
+            releases: cleanReleases,
+            releaseTypes,
+            releaseTags,
+          } as FreshReleasesData,
+          hasError: false,
+          errorMessage: "",
+        };
       } catch (error) {
         toast.error(
           <ToastMsg
             title="Couldn't fetch fresh releases"
-            message={
-              typeof error === "object" ? error.message : error.toString()
-            }
+            message={error.message}
           />,
           { toastId: "fetch-error" }
         );
+        return {
+          data: {
+            releases: [],
+            releaseTypes: [],
+            releaseTags: [],
+          } as FreshReleasesData,
+          hasError: true,
+          errorMessage: error.message,
+        };
       }
-    };
-    // Call the async function defined above (useEffect can't return a Promise)
-    fetchReleases();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [pageType, range, showPastReleases, showFutureReleases, sort]);
+    },
+  });
+
+  const {
+    data: rawData = {
+      releases: [],
+      releaseTypes: [],
+      releaseTags: [],
+    } as FreshReleasesData,
+    hasError = false,
+    errorMessage = "",
+  } = loaderData || {};
+
+  const { releases, releaseTypes, releaseTags } = rawData;
+
+  const releasesString = JSON.stringify(releases);
+  React.useEffect(() => {
+    const newReleases = JSON.parse(releasesString);
+    setFilteredList(newReleases);
+  }, [releasesString]);
```

Comment:
I'm not sure how to test this locally properly, but I wonder if you need the stringification here.
Does this result int he same re-rendering loop issue still?
If so, I think something smells a bit fishy with the way we manipulate the state between parent and child components...
```
React.useEffect(() => {
    setFilteredList(releases);
  }, [releases]);
```

Reply:
Yes, If i don't add a stringification, I get a warning on the browser console

`FreshReleases.tsx:283 Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.`

Reply:
I found the culprit !
I did say something was smelling fishy :D

In cases of circular dependency/render issues like this, my usual first step is to look at the dependency arrays and see if anything is missing in them.
In this case that turned out o be the issue, so an easy fix.
Disabling the linting rule `react-hooks/exhaustive-deps` is a dangerous game, it turns out :P

In our case, since we had `if (!_.isEqual(filteredReleases, filteredList)) {` and `filteredList` was never updated in the effect in ReleaseFilters, the condition was always true and was calling the parent's `setFilteredList`, which in turn re-rendered the child with the new `filteredList`, and so on, hence the infinite re-render loop.

Reply:
Got it. Thank you so much for fixing this!